### PR TITLE
Rename classes to match their Qt Quick counterparts

### PR DESCRIPTION
**Pull request by 0x6e**
DeclarativeDeclarativeContext -> DeclarativeQmlContext
DeclarativeDeclarativeViewExtension -> DeclarativeQuickWidgetExtension

### CODE REVIEW

**Comment by 0x6e** [comment]
Ah, this isn't ready for review yet. I've messed something up with my rebasing.

**Comment by 0x6e** [comment]
This is now ready to review.

File: lib/declarativewidgetsdocument.cpp [hunk 1]
```diff
@@ -158,7 +158,7 @@ DeclarativeWidgetsDocument::DeclarativeWidgetsDocument(const QUrl &url, QObject
   qmlRegisterExtendedType<QComboBox, DeclarativeComboBoxExtension>("QtWidgets", 1, 0, "ComboBox");
   qmlRegisterExtendedType<QDateEdit, DeclarativeWidgetExtension>("QtWidgets", 1, 0, "DateEdit");
   qmlRegisterExtendedType<QDateTimeEdit, DeclarativeWidgetExtension>("QtWidgets", 1, 0, "DateTimeEdit");
-  qmlRegisterExtendedType<QQuickWidget, DeclarativeDeclarativeViewExtension>("QtWidgets", 1, 0, "DeclarativeView");
+  qmlRegisterExtendedType<QQuickWidget, DeclarativeQuickWidgetExtension>("QtWidgets", 1, 0, "DeclarativeView");
```
Comment (krake-kdab) [inline lib/declarativewidgetsdocument.cpp]: I would also change the QML name to "QuickWidget"

File: tests/tst_declarativeview.cpp [hunk 2]
```diff
@@ -42,9 +42,9 @@ private slots:
 tst_DeclarativeView::tst_DeclarativeView()
 {
     qmlRegisterType<DeclarativeContextProperty>("Qt.Widgets", 1, 0, "DeclarativeContextProperty");
-    qmlRegisterType<DeclarativeDeclarativeContext>("Qt.Widgets", 1, 0, "DeclarativeContext");
+    qmlRegisterType<DeclarativeQmlContext>("Qt.Widgets", 1, 0, "DeclarativeContext");
 
-    qmlRegisterExtendedType<QQuickWidget, DeclarativeDeclarativeViewExtension>("Qt.Widgets", 1, 0, "DeclarativeView");
+    qmlRegisterExtendedType<QQuickWidget, DeclarativeQuickWidgetExtension>("Qt.Widgets", 1, 0, "DeclarativeView");
```
Comment (krake-kdab) [inline tests/tst_declarativeview.cpp]: same here

**Pull request merged**